admin/class-admin-apple-notice.php

```diff
@@ -430,4 +430,23 @@ private static function update_user_meta( $user_id, $values ) {
 
 		return true;
 	}
+
+	/**
+	 * Checks if the current user has access to see the notifications and returns them if true.
+	 *
+	 * @param array $object The requested object.
+	 * @return array
+	 */
+	public static function get_if_allowed( $object ) {
+		$post_type = ( $object['type'] );
+		if ( current_user_can(
+			apply_filters( 'apple_news_publish_capability', Apple_News::get_capability_for_post_type( 'publish_posts', $post_type ) )
+		) ) {
+			$notifications = self::get();
+			self::clear( $notifications );
+		} else {
+			$notifications = [];
+		}
+		return $notifications;
+	}
```

Comment:
We want to bail out of this as early as possible with as light of a touch as possible, since this will run on every REST request. Here are some suggestions:
```suggestion
	public static function get_if_allowed( $object ) {
		$notifications = [];

		// Ensure we have an object type.
		if ( empty( $object['type'] ) ) {
			return $notifications;
		}
		
		// This functionality is only available to logged in users.
		if ( ! is_user_logged_in() ) {
			return $notifications;
		}
		
		// Ensure current user has the appropriate publish permission.
		if ( ! current_user_can(
			apply_filters( 'apple_news_publish_capability', Apple_News::get_capability_for_post_type( 'publish_posts', $object['type'] ) )
		) ) {
			return $notifications;
		}
		
		// Get notifications.
		$notifications = self::get();
		self::clear( $notifications );

		return $notifications;
	}
```